ssl/statem/extensions.c

```diff
@@ -1427,10 +1427,18 @@ int tls_psk_do_binder(SSL *s, const EVP_MD *md, const unsigned char *msgstart,
     const char external_label[] = "ext binder";
     const char nonce_label[] = "resumption";
     const char *label;
-    size_t bindersize, labelsize, psklen, hashsize = EVP_MD_size(md);
+    size_t bindersize, labelsize, psklen, hashsize;
+    int hashsizei = EVP_MD_size(md);
     int ret = -1;
     int usepskfored = 0;
 
+    if (!ossl_assert(hashsizei >= 0)) {
```

Comment:
Formally speaking, from cast safety viewpoint, it's correct. But if you follow hashsize's usage you'll note that it describes length of data placed into EVP_MAX_MD_SIZE-ed objects. I.e. it means that it may not be larger than EVP_MAX_MD_SIZE. And one can wonder if it actually may be 0. However! This is not an argument in favour of making this check more elaborate. The *only* thing I'm saying is that if you read the code in its entirety it asks for questions. In other words it's readability issue. And maybe `/* ensure cast to size_t is safe */` comment would be appropriate. Alternatively one can wonder if check for md being non-NULL would be adequate (naturally prior calling EVP_MD_size)... This is obviously a side note...